### PR TITLE
Update AoC leaderboard upload ot use new paste service

### PR DESCRIPTION
**Pull request by ChrisLovering**

### CODE REVIEW

File: bot/exts/advent_of_code/_helpers.py [hunk 1]
```diff
@@ -289,18 +287,20 @@ async def _fetch_leaderboard_data() -> dict[str, Any]:
...
-    log.error(f"Unexpected response from paste service while uploading leaderboard {resp_json}")
-    return ""
+    file = paste_service.PasteFile(content=leaderboard)
+    try:
+        paste = await paste_service.send_to_paste_service(
+            files=[file],
+            http_session=aiohttp.ClientSession()
+        )
+    except paste_service.PasteTooLongError:
+        log.exception("Leaderboard too long to upload to paste site")
```
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/exts/advent_of_code/_helpers.py:297]: log.exception automatically grabs exception context when it's used inside an except block

**Review by shtlrs — APPROVED** [review]
Didn't test this, but the change looks safe and good to me.

File: bot/exts/advent_of_code/_helpers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChrisLovering) [reply bot/exts/advent_of_code/_helpers.py:297]: To be more specific, log.exception can only be used in an exception block, and it is a shortcut to `log.error("foo", exc_info=err)` where it automatically grabs exec info using `sys.exc_info()`

**Comment by janine9vn** [comment]
I'm getting this as an error when I test it. Is it something about my settings/config that's wrong?
image

**Comment by ChrisLovering** [comment]
> I'm getting this as an error when I test it. Is it something about my settings/config that's wrong?

You're probably not doing anything wrong. I didn't get a chance to test it when I made this, then quickly forgot about it.

Will give it a look later today.

**Review by janine9vn — APPROVED** [review]
Tested and now works with the top 1000 limit. Maybe in many years we'll have to deal with more than 1,000 active participants but that's a very-future-us problem.

**Pull request merged**